cypress-io/cypress
# fix: support custom workspaceRoot for angular CT projectConfig

### User facing changelog

angular CT will respect a `workspaceRoot` build option if provided, which is useful for when cypress opens in a nested folder. i.e. an nx workspace

### Additional details

When using Angular component testing within an nx workspace, assets will not load as the proxied path was incorrectly set since Cypress opens within a nested folder. 

### Steps to test

> This change does requires changes to the @nrwl/angular nx component testing preset. Mainly the offset value is removed and a custom 'workspaceRoot' is passed to cypress. This can be done within node_modules if really wanted to test it works before this change is in Nx

1. make a new nx angular workspace, `npx create-nx-workspace@latest --preset=angular`
2. add an asset to the apps asset folder, `apps/<your-app-name>/src/assets`
3. load the asset in the app component, i.e. `<img src="assets/<your-asset> />`
4. add component testing to the app, `npx nx g cypress-component-configuration`
5. test the component that's loading the asset
6. Asset should load

### How has the user experience changed?

Doesn't change the current experience, only improves for those who might need custom sets (like Nx :) )

### PR Tasks
